src/lib/ppx_version/lint_version_syntax.ml

```diff
@@ -1,14 +1,24 @@
-(* lint_version_syntax.ml -- static enforcement of syntactic items relating to proper versioning
+(* lint_version_syntax.ml -- static enforcement of syntactic items relating to proper versioning *)
+
+(* The lint rules:
+
+   - "deriving bin_io" and "deriving version" never appear in types defined inside functor bodies,
+     except for `Make_str` functors used for wire types.
 
-   - "deriving bin_io" and "deriving version" never appear in types defined inside functor bodies
    - otherwise, "bin_io" may appear in a "deriving" attribute only if "version" also appears in that extension
+
    - versioned types only appear in versioned type definitions
+
    - versioned type definitions appear only in %%versioned... extensions
+
    - packaged modules, like "(module Foo)", may not be stable-versioned (but allowed inside %%versioned for
        legitimate uses)
```

Comment:
```suggestion
   - otherwise, "bin_io" might appear in a "deriving" attribute only if "version" also appears in that extension

   - versioned types appear only in versioned type definitions

   - versioned type definitions appear only in %%versioned... extensions

   - packaged modules, like "(module Foo)", might not be stable-versioned (but allowed inside %%versioned for
       legitimate uses)
```

Reply:
may is permissions https://developers.google.com/style/word-list#may
let's use might https://developers.google.com/style/word-list#might

Reply:
in this case, "may" is indicating policy, which is enforced by the linter